File: client-android/lib/src/main/kotlin/stasis/client_android/lib/model/server/api/requests/CreateAnalyticsEntry.kt
package stasis.client_android.lib.model.server.api.requests

import com.squareup.moshi.JsonClass
import stasis.client_android.lib.telemetry.analytics.AnalyticsEntry

@JsonClass(generateAdapter = true)
data class CreateAnalyticsEntry(
    val entry: AnalyticsEntry.AsJson
)

File: shared/src/main/scala/stasis/shared/api/requests/CreateAnalyticsEntry.scala
package stasis.shared.api.requests

import java.time.Instant

import io.github.sndnv.layers.telemetry.analytics.AnalyticsEntry
import stasis.shared.model.analytics.StoredAnalyticsEntry

final case class CreateAnalyticsEntry(entry: AnalyticsEntry) {
  def toStoredAnalyticsEntry: StoredAnalyticsEntry =
    StoredAnalyticsEntry(
      id = StoredAnalyticsEntry.generateId(),
      runtime = entry.runtime,
      events = entry.events,
      failures = entry.failures,
      created = entry.created,
      updated = entry.updated,
      received = Instant.now()
    )
}
